sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -36,20 +36,29 @@ internal static List<TelemetryItem> OtelToAzureMonitorTrace(Batch<Activity> batc
                     var activityTagsProcessor = EnumerateActivityTags(activity);
                     telemetryItem = new TelemetryItem(activity, ref activityTagsProcessor, azureMonitorResource, instrumentationKey);
 
-                    // Check for Exceptions events
-                    if (activity.Events.Any())
-                    {
-                        AddTelemetryFromActivityEvents(activity, telemetryItem, telemetryItems);
-                    }
-
                     switch (activity.GetTelemetryType())
                     {
                         case TelemetryType.Request:
-                            telemetryItem.Data = new MonitorBase
+                            if (activity.Kind == ActivityKind.Server)
                             {
-                                BaseType = "RequestData",
-                                BaseData = new RequestData(Version, activity, ref activityTagsProcessor)
-                            };
+                                var (requestUrl, operationName) = GetHttpOperationNameAndUrl(activity.DisplayName, activityTagsProcessor.activityType, ref activityTagsProcessor.MappedTags);
+                                telemetryItem.Tags[ContextTagKeys.AiOperationName.ToString()] = operationName;
```

Comment:
@TimothyMothra We may need a follow up PR here to check if this requires Truncation. Truncation was not a part of existing implementation.

Reply:
`Tags` don't have a max length defined in the swagger.
https://github.com/Azure/azure-rest-api-specs/blob/55fc71748c6f51a049fb77aa162ca797aabfbae6/specification/applicationinsights/data-plane/Monitor.Exporters/preview/v2.1/swagger.json#L872C18-L872C18

Reply:
I found the length for AIOperationName. It's in an internal repo. I'll get this updated